scripts/CSSRE/lib/py/kafka.py

```diff
@@ -0,0 +1,215 @@
+from py.logger import Logger
+from py.exceptions import StatefulsetExecConnectionError
+import copy
+
+class Kafka:
+  """
+  A Kafka utility class for getting information about Kafka clusters
+  """
+  def __init__(self, oc, settings, logger_name=__name__):
+    self._oc = oc
+    self.settings = settings
+    self.logger = Logger.get_logger(logger_name)
+
+  @property
+  def oc(self): 
+      return self._oc
+
+
+  def get_kafka_cluster(self, namespace):
+    """
+    Gets Kafka cluster in a given namespace.
+
+    Parameters:
+      namespace (str): Namespace the get Kafka cluster.
+
+    Returns:
+      (apiobject): OpenShift API object for the Kafka cluster.
+
+    """
+    with self._oc.project(namespace), self._oc.timeout(self.settings.OC_TIMEOUT_5_MINUTES):
+      return next(iter(self._oc.selector('kafkas').objects() or []), None)
+
+
+  def check_all_brokers_active(self, namespace):
+    """
+    Check if all brokers for a Kafka are ready.
+
+    Parameters:
+      namespace (str): Namespace the get Kafka cluster.
+
+    Returns:
+      (int))num_pods_ready: Number of ready brokers. 
+      (int)pods: Total nuymber of brokers.
+
+    """
+    with self._oc.project(namespace), self._oc.timeout(self.settings.OC_TIMEOUT_5_MINUTES):
+      self.logger.debug("Checking if Kafka pods are active")
+      pods = self._oc.selector('pods', labels={"app.kubernetes.io/name": "kafka"}).objects()
+      num_pods_ready = 0
+      for pod in pods:
+        ready = self._oc.selector('pod/'+pod.name()).object().model.status.conditions.can_match(
+          {
+              'type': 'Ready',
+              'status': "True",
+          }
+        )
+        if ready:
+          num_pods_ready +=1
+        self.logger.debug("Kafka Pod %s ready: %s." % (pod.name(), ready))
+    return (num_pods_ready, len(pods))
+
+
+  def run_kafka_topics_script(self, namespace, kafka, topic="", filter="") :
+    """
+    Runs the kafka-topics.sh script on a kafka pod in a given namespace.
+
+    Parameters:
+      namespace (str): Namespace of the kafka.
+      kafka (str): Name of the kafka cluster.
+      filter (str): a filter for partitions (--under-replicated-partitions, --under-min-isr-partitions, --unavailable-partitions).
+
+    Returns
+      (apiobject): OpenShift API object with description of topics and partitions.
+    """
+    with self._oc.project(namespace), self._oc.timeout(self.settings.OC_TIMEOUT_5_MINUTES):
+
+      cmd = "-i statefulset/"+kafka+"-kafka -c kafka -- env - bin/kafka-topics.sh --bootstrap-server localhost:9096 --describe "
+      if topic:
+        cmd = cmd + " --topic " + topic
+      if filter:
+        cmd = cmd + " --" + filter
+      try:
+        return self._oc.invoke('exec', cmd.split())
+      except self._oc.OpenShiftPythonException as e:
```

Comment:
here's the warning where the pipeline failed, `e`

Reply:
This var can be removed.